### PR TITLE
feat: adding rocket chat in right side panel in video room

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes #add rocket chat in side panel in videoroom

Screenshot from 2021-04-01 18-28-18

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mariobehling** [comment]
Thanks, I tested this. Looks good. Please change the following.

1. Instead of the "<" arrow, show a chat icon on the bottom right of the page. Something like intercom.com or tawk.to.
Screenshot from 2021-04-09 09-31-31
2. To close the side panel show an X on the top left of the side panel itself.
3. Also show the side panel on the other pages, e.g. exhibition, speaker page etc.

**Comment by maze-runnar** [comment]
Screenshot from 2021-04-10 11-31-33
Screenshot from 2021-04-10 11-31-25
Screenshot from 2021-04-10 11-08-14
Screenshot from 2021-04-10 11-08-05

File: app/routes/public/speakers.js [hunk 1]
```diff
@@ -65,7 +65,10 @@ export default class SpeakersRoute extends Route {
         ]
       });
     }
+    const { success, token } = await this.loader.load(`/events/${eventDetails.id}/chat-token`);
```
Comment (iamareebjamal) [inline app/routes/public/speakers.js:68]: Wrong logic, first this is duplication of code. Secondly, this should be fetched in the component. That's why it is logging in user before opening the chat

File: app/routes/public/sessions.js [hunk 2]
```diff
@@ -197,8 +197,10 @@ export default class SessionsRoute extends Route {
         }
       });
     }
-
+    const { success, token } = await this.loader.load(`/events/${eventDetails.id}/chat-token`);
```
Comment (iamareebjamal) [inline app/routes/public/sessions.js:200]: Wrong logic, first this is duplication of code. Secondly, this should be fetched in the component. That's why it is logging in user before opening the chat

File: app/routes/public/exhibition/index.js [hunk 3]
```diff
@@ -20,7 +20,10 @@ export default class ExhibitionRoute extends Route {
         val  : 'accepted'
       }
     ];
+    const { success, token } = await this.loader.load(`/events/${eventDetails.id}/chat-token`);
```
Comment (iamareebjamal) [inline app/routes/public/exhibition/index.js:23]: Wrong logic, first this is duplication of code. Secondly, this should be fetched in the component. That's why it is logging in user before opening the chat

**Comment by maze-runnar** [comment]
Screenshot from 2021-04-12 09-02-36

**Comment by maze-runnar** [comment]
Screenshot from 2021-04-13 20-04-22
Screenshot from 2021-04-13 20-01-23
Screenshot from 2021-04-13 20-01-04

**Comment by iamareebjamal** [comment]
On public page, it is also shown if user is logged out. It should only be shown if user is logged in and has access to the streams. Like the join live event button dialog. 

Also, move the button a bit down on the page and the cross button on side panel a bit up

**Comment by maze-runnar** [comment]
Screenshot from 2021-04-13 22-05-43
Screenshot from 2021-04-13 22-05-48

**Pull request merged**